pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                            .then(d => {
+                                this.setState((state, _) => {
+                                    const installed = (d.unavailable_names || []).length === 0 && (d.missing_names || []).length === 0;
+                                    const unavailable = (d.unavailable_names || []).length > 0;
+                                    const current = state.enabled && (installed || unavailable);
+                                    return ({
+                                        patchName: patch_name,
+                                        patchInstalled: installed,
+                                        patchUnavailable: unavailable,
+                                        justCurrent: current && !state.auto,
+                                        applyCheckbox: current,
+                                        kernelName: data,
+                                    });
+                                });
+                            })
+                            .catch(() => true);
+                })
+                .catch(() => true);
+    }
+
+    handleInstall() {
+        this.setState({ updating: true });
+        install_dialog(this.state.missing)
+                .then(() => {
+                    this.setState({ missing: [], updating: false });
+                })
+                .catch(() => this.setState({ updating: false }));
+    }
+
+    onClose() {
+        this.setState((state, _) => {
+            const current = state.enabled && (state.patchInstalled || state.patchUnavailable);
+            return ({
+                justCurrent: current && !state.auto,
+                applyCheckbox: current,
+                showModal: false,
+                error: "",
+            });
+        });
+    }
+
+    handleChange() {
+        this.setState({ updating: true });
+
+        if (this.state.applyCheckbox) {
+            let install = Promise.resolve(true);
+            if (this.state.justCurrent) {
+                install = new Promise((resolve, reject) => {
+                    cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" })
+                            .then(() => {
+                                if (!this.state.patchUnavailable && !this.state.patchInstalled)
+                                    cockpit.spawn(["dnf", "-y", "install", this.state.patchName], { superuser: "require", err: "message" }).then(resolve)
+                                            .catch(reject);
+                                else
+                                    resolve();
+                            })
+                            .catch(reject);
+                });
+            } else {
+                install = cockpit.spawn(["dnf", "-y", "kpatch", "auto"], { superuser: "require", err: "message" });
+            }
+            install.then(() => {
+                this.kpatchService.enable().then(() => {
```

Comment:
Please drop the { } wrappers or `return` the next promise. Like this it ignores failures of the inner promise as they are not properly chained. (Same below in the "else" branch)